pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -277,6 +288,7 @@ public String getTaskType() {
         }
 
         Long prevWatermarkMs = mergeRollupTaskMetadata.getWatermarkMap().put(mergeLevel, windowStartMs);
+        setWatermarkMs(offlineTableName, mergeLevel, waterMarkMs, bufferMs, bucketMs);
```

Comment:
The third argument should be `windowStartMs` instead of `watermarkMs`. `watermarkMs` is the from the previous run, which might be lower than the `windowStartMs`

Reply:
I got confused due to the following line from the log. I updated the log as well to correctly log old & new watermark. The code did not correctly log the info like the following: 
```
Update watermark of mergeLevel: 1day for table: xxx from: 1632700800000 to: 1632700800000
Update watermark of mergeLevel: 1day for table: xxx from: 1632700800000 to: 1632700800000
```